### PR TITLE
Perl bindings

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This is a continuation/replacement of #5 and builds on @kenodegard's work there.

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
So... this finally seems to work. The key was in moving the installation destinations for the perl bindings from the "normal"(?) `$PREFIX/lib/site_perl/5.32.1/x86_64-linux-thread-multi` to `$PREFIX/lib/perl5/5.32/site_perl` by passing some arguments to makemakers `Makefile.PL`.

I believe this is due to the way things are set up in https://github.com/conda-forge/perl-feedstock/blob/1b83a674d79e6df7bdd8a16c4aa6b051467a41a0/recipe/build.sh#L22-L27, but I would be more than happy if someone from @conda-forge/perl could look over this.

@kenodegard, do you also want to have a look? Do you want to adopt this in #5, or should we move forward with this PR?

This could unblock conda-forge/git-feedstock#60 and thus bring git-svn to conda-forge, which is what I am really after :wink:

**Comment by kenodegard** [comment]
@zklaus I unfortunately have long since moved past this struggle (yes, I too was trying to get git-svn to work) but I know of people who would greatly appreciate this fix! If you got this to work here I would encourage proceeding with this one and closing out #5.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -39,15 +39,19 @@ requirements:
     - cyrus-sasl  # [unix]
     - libmagic  # [unix]
     - serf  # [linux]
+    - swig
+    - perl >=5.8.0
```
Comment (isuruf) [inline recipe/meta.yaml]: To use the conda-forge global pinning

```suggestion
    - perl
```

**Comment by jakirkham** [comment]
cc @mbargull (as you have been doing work on `perl` and may have thoughts here)

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
> @zklaus I unfortunately have long since moved past this struggle (yes, I too was trying to get git-svn to work) but I know of people who would greatly appreciate this fix! If you got this to work here I would encourage proceeding with this one and closing out #5.

Cheers. Just wanted to make sure I am not stepping on your toes. Would you mind closing #5 then? Since I am not a maintainer yet, I can't do that.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
Ok, sorry for some confusion, here is an update.
With the pinning removed (thanks @isuruf!), the global pinning came into force, and that lead to a downgrade of perl from 5.32.1-1_h7f98852_perl5 to 5.26.2-h36c2ea0_1008.

However, the global pinning is not applied to the build environment, possibly because perl is only listed as a host, not as a build requirement. But because perl is still also installed in the build environment, there it ends up as version 5.32.1, which is the wrong version to build for.

The way I have resolved this now is by specifying explicitly `${PREFIX}/bin/perl` in the call to MakeMaker, but I wonder if this is correct.

Is it? Or should perl be added to the build requirements to apply global pinning also there?

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
I have convinced myself that this is the right approach. Perl bindings are only built for Linux; the Macos build produces a segfault on the test and will be investigated further later on.

I believe this is ready for merge.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
With the recently finished Perl migration, the preferred way of installing Perl modules seems to have changed slightly. With this in mind, let's rerender and give it another try.

**Comment by zklaus** [comment]
Ok, now it also works with macos. Anybody up for a merge?

**Comment by jakirkham** [comment]
Thanks @zklaus! 😄

**Pull request merged**